instrumentation/akka/akka-http-10.0/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/akkahttp/AkkaHttpClientInstrumentationTest.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.akkahttp
+
+import akka.actor.ActorSystem
+import akka.dispatch.ExecutionContexts
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.javadsl.model.HttpHeader
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.stream.ActorMaterializer
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension
+import io.opentelemetry.instrumentation.testing.junit.http.{
+  AbstractHttpClientTest,
+  HttpClientInstrumentationExtension,
+  HttpClientTestOptions,
+  SingleConnection
+}
+import java.net.URI
+import java.util
+import java.util.concurrent.Executor
+import org.junit.jupiter.api.extension.RegisterExtension
+import scala.compat.java8.FunctionConverters._
+import scala.collection.JavaConverters._
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+class AkkaHttpClientInstrumentationTest
+    extends AbstractHttpClientTest[HttpRequest] {
+
+  @RegisterExtension val _: InstrumentationExtension =
+    HttpClientInstrumentationExtension.forAgent()
+
+  val system: ActorSystem = ActorSystem.create()
+  val materializer: ActorMaterializer = ActorMaterializer.create(system)
+
+  override protected def buildRequest(
+      method: String,
+      uri: URI,
+      h: util.Map[String, String]
+  ): HttpRequest =
+    HttpRequest(HttpMethods.getForKey(method).get)
+      .withUri(uri.toString)
+      .addHeaders(
```

Comment:
Not a big deal but no matter how much I rattled I couldn't get the scaladsl version of this (`withHeaders`) to work, I was expecting `toSeq` instead of `asJava` to work. In case anyone wants to test their scala chops :P